packages/server-core/src/media/static-resource/static-resource.hooks.ts

```diff
@@ -103,6 +103,20 @@ const createHashIfNeeded = async (context: HookContext<StaticResourceService>) =
   }
 }
 
+const updateName = async (context: HookContext<StaticResourceService>) => {
+  if (!context.data || !(context.method === 'create' || context.method === 'update' || context.method === 'patch')) {
+    throw new BadRequest(`${context.path} service only works for data in ${context.method}`)
+  }
+
+  if (Array.isArray(context.data)) throw new BadRequest('Batch create is not supported')
+
+  const data = context.data
+  if (!data.key) return
+
+  const [_, directory, file] = /(.*)\/([^\\\/]+$)/.exec(data.key)!
+  context.data.name = file
```

Comment:
we need to add some more logic here

- always set the name on create method
- if update or patch method, and a static resource entry already exists, check to see if the name matches the name part of  key
  - update the name derived from the key, otherwise do not update the name (this will preserve custom asset names)

Reply:
implemented